internal/reconcile/atomic_release.go

```diff
@@ -278,21 +290,23 @@ func (r *AtomicRelease) Reconcile(ctx context.Context, req *Request) error {
 					"instructed to stop after running %s action reconciler %s", next.Type(), next.Name()),
 				)
 
+				if retry := req.Object.GetActiveRetry(); retry != nil {
+					conditions.Delete(req.Object, meta.ReconcilingCondition)
+					conditions.MarkFalse(req.Object, meta.ReadyCondition, "RetryAfterInterval",
+						"Will retry %s after %s", retry.GetAction(), retry.GetRetryInterval().String())
+					return ErrRetryAfterInterval
+				}
+
 				remediation := req.Object.GetActiveRemediation()
 				if remediation == nil || !remediation.RetriesExhausted(req.Object) {
 					conditions.MarkReconciling(req.Object, meta.ProgressingWithRetryReason, "%s", conditions.GetMessage(req.Object, meta.ReadyCondition))
 					return ErrMustRequeue
 				}
-				// Check if retries have exhausted after remediation for early
-				// stall condition detection.
-				if remediation != nil && remediation.RetriesExhausted(req.Object) {
```

Comment:
This check is a tautology, it's basically the negation of the check right above. I removed it in this separate commit: https://github.com/fluxcd/helm-controller/pull/1281/commits/079ae1b7e7a6e307a29b582afb095323749ac5c6